### PR TITLE
Add aliases by hand

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR makes it possible to add aliases by hand.

### CODE REVIEW

File: static/js/components/AddGcnAlias.jsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  },
+  editIcon: {
+    height: "0.75rem",
+    cursor: "pointer",
+  },
+}));
+
+const AddGcnAlias = ({ gcnEvent }) => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
```
Comment (Theodlz) [inline static/js/components/AddGcnAlias.jsx:30]: Frontend looks good! Nice

File: skyportal/handlers/api/gcn.py [hunk 2]
```diff
@@ -489,6 +490,147 @@ def post_gcnevent_from_dictionary(payload, user_id, session, asynchronous=True):
     return event.id
 
 
+class GcnEventAliasesHandler(BaseHandler):
+    @auth_or_token
+    def post(self, dateobs):
+        """
+        ---
+        description: Post a GCN Event tag
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/gcn.py:498]: Good catch.

File: skyportal/handlers/api/gcn.py [hunk 3]
```diff
@@ -489,6 +490,147 @@ def post_gcnevent_from_dictionary(payload, user_id, session, asynchronous=True):
...
+        tags:
+          - gcnevents
+        parameters:
+          - in: path
+            name: dateobs
+            required: true
+            schema:
+              type: string
+            description: The dateobs of the event, as an arrow parseable string
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/gcn.py:507]: That's listed below.

File: skyportal/handlers/api/gcn.py [hunk 4]
```diff
@@ -489,6 +490,147 @@ def post_gcnevent_from_dictionary(payload, user_id, session, asynchronous=True):
...
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        data = self.get_json()
+        alias = data.get('alias', None)
+
+        if alias is None:
+            return self.error("alias must be present in data")
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/gcn.py:533]: ah good.

File: skyportal/handlers/api/gcn.py [hunk 5]
```diff
@@ -489,6 +490,147 @@ def post_gcnevent_from_dictionary(payload, user_id, session, asynchronous=True):
...
+        ---
+        description: Delete a GCN event alias
+        tags:
+          - gcnevents
+        parameters:
+          - in: path
+            name: dateobs
+            required: true
+            schema:
+              type: dateobs
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/gcn.py:575]: It is in the requestBody.

File: skyportal/handlers/api/gcn.py [hunk 6]
```diff
@@ -489,6 +490,147 @@ def post_gcnevent_from_dictionary(payload, user_id, session, asynchronous=True):
...
+                    ).where(GcnEvent.dateobs == dateobs)
+                ).first()
+                if event is None:
+                    return self.error("GCN event not found", status=404)
+
+                if alias in event.aliases:
+                    aliases = event.aliases
+                    aliases.remove(alias)
+                    setattr(event, 'aliases', aliases)
+                    flag_modified(event, 'aliases')
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/gcn.py:619]: yes adding.

File: skyportal/handlers/api/gcn.py [hunk 7]
```diff
@@ -489,6 +490,147 @@ def post_gcnevent_from_dictionary(payload, user_id, session, asynchronous=True):
...
+              application/json:
+                schema: Error
+        """
+
+        data = self.get_json()
+        alias = data.get('alias')
+
+        if alias is None:
+            return self.error("alias must be present in data to remove")
+
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/gcn.py:605]: yes.

**Comment by Theodlz** [comment]
@mcoughlin I'm wondering if this last_modified thing might have broken the other feature. Weird that its required for one and not the other... Tests are failing on main. I'll open a PR super soon

**Comment by mcoughlin** [comment]
@Theodlz I tested and added an alias, and then ingested normally. It seemed fine.

**Comment by Theodlz** [comment]
Hmmm. Maybe just a TACH issue then. I'll check

> @Theodlz I tested and added an alias, and then ingested normally. It seemed fine.

**Pull request merged**